eng/native/gen-buildsys.sh

```diff
@@ -104,16 +104,14 @@ if [[ "$host_arch" == "wasm" ]]; then
     fi
 fi
 
-# We have to be able to build with CMake 3.6.2, so we can't use the -S or -B options
-pushd "$2"
-
 $cmake_command \
   --no-warn-unused-cli \
   -G "$generator" \
   "-DCMAKE_BUILD_TYPE=$buildtype" \
   "-DCMAKE_INSTALL_PREFIX=$__CMakeBinDir" \
   $cmake_extra_defines \
   $__UnprocessedCMakeArgs \
-  "$1"
+  -S "$1" \
+  -B "$3"
 
-popd
+exit $?
```

Comment:
still keeping this so it's clearer for the next person touching this line that the exit code is important